articles/python-static-typing/index.md

```diff
@@ -0,0 +1,218 @@
+Python is a dynamically typed language. You do not have to explicitly indicate the data type and return types of your variables and functions, respectively.
+Dynamic typing makes Python very user-friendly. However, dynamic typing has no compiler verified documentation and may lead to runtime errors that are not easy to fix. Static typing provides a solution to these problems.
+This article will see how to perform static typing in Python and handle both variable and function annotations. You need to have a basic understanding of Python to follow along. You will also install [mypy](http://mypy-lang.org/) for type checking.
+
+### Table of contents
+
+1. Introduction to type annotations
+2. Variable annotations
+3. Function annotations
+4. Selected annotations in Python `Typing` module
+   - Callable
+   - List
+   - Dict & Any
+   - Union
+5. Using mypy static type checker
+
+### 1. Introduction to type annotations
+
+Type annotations is a new feature added by [PEP 484](https://www.python.org/dev/peps/pep-0484/) to python version 3.5 and above. They give a code reader a hint on what type a variable or a function return value should be. Type annotations can be performed on variables and functions as well.
+
+### 2. Variable annotations
+
+We achieve variable annotation by adding a semicolon and the type after declaring or initializing a variable. i.e `: <type>`. Like in the example below, instead of saying `name = "John Doe"` we specify the return type by writing `name: str = "John Doe"`
+
+```python
+name: str = "John Doe"
+print(name)
+
+```
+
+### 3. Function annotations
+
+This is achieved by adding a forward arrow and indicating the function parameters' expected return type after the function parameters' closing bracket. i.e `-> <return type>`.
+
+```python
+
+def square(x: int) -> int:
+    return x * x
+
+```
```

Comment:
Mention the output of the code as well. Do it for all code snippets in the article.